src/wire/packet_manager.cpp

```diff
@@ -123,16 +124,43 @@ void PacketManager::MakeHardcodedParameterStatus(
   PacketPutString(response.get(), kv.second);
   responses.push_back(std::move(response));
 }
+
 /*
  * process_startup_packet - Processes the startup packet
  *  (after the size field of the header).
  */
-bool PacketManager::ProcessStartupPacket(InputPacket *pkt) {
+int PacketManager::ProcessInitialPacket(InputPacket *pkt) {
   std::string token, value;
   std::unique_ptr<OutputPacket> response(new OutputPacket());
 
   int32_t proto_version = PacketGetInt(pkt, sizeof(int32_t));
+  LOG_INFO("protocol version: %d", proto_version);
+  bool res;
+  int res_base = 0;
+  // TODO: consider more about return value
+  if (proto_version == 80877103) {
```

Comment:
Is there any better place to put `80877103` in? I understand this magic number is necessary here. But it might be more readable if you could define a macro or const int for it. And maybe you could provide more information about why this version number is special?